ckanext/s3filestore/uploader.py

```diff
@@ -219,6 +222,46 @@ def upload(self, max_size=2):
                 and not self.old_filename.startswith('http')):
             self.clear_key(self.old_filepath)
 
+    def delete(self):
+        ''' Delete file we are pointing at'''
+        key_path = self.filepath
+        try:
+            self.clear_key(key_path)
+        except ClientError as ex:
+            log.warning('Key \'{0}\' not found in bucket \'{1}\' for delete'
+                        .format(key_path, self.bucket_name))
+            pass
+
+    def download(self):
+        '''
+        Provide a download by either redirecting the user to the url stored or
+        downloading the uploaded file from S3.
+        '''
+
+
+        key_path = self.filename()
+
+        if key_path is None:
+            log.warning('Key \'{0}\' not found in bucket \'{1}\''
+                     .format(key_path, self.bucket_name))
+
+        try:
+            # Small workaround to manage downloading of large files
```

Comment:
We could probably extract parts of this code to a helper function since it's now repeated three times IIUC.